src/libraries/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -182,7 +182,17 @@ public void GetTempFileName()
             try
             {
                 Assert.True(File.Exists(tmpFile));
-                Assert.Equal(".tmp", Path.GetExtension(tmpFile), ignoreCase: true, ignoreLineEndingDifferences: false, ignoreWhiteSpaceDifferences: false);
+                string fileName = Path.GetFileName(tmpFile);
+                Assert.StartsWith("tmp", fileName);
+                Assert.Equal("tmpXXXXXX.tmp".Length, fileName.Length);
+                Assert.EndsWith(".tmp", fileName);
+
+                const string validChars = "abcdefghijklmnopqrstuvwxyz0123456789";
+                for (int i = 3; i < 9; i++)
+                {
+                    Assert.True(validChars.Contains(char.ToLowerInvariant(fileName[i])));
```

Comment:
The char.ToLowerInvariant here is to avoid adding "ABCDEFGHIJKLMNOPQRSTUVWXYZ" to the above const? Maybe reconsider :)